pkg/operation/etcdencryption/encryptionkeys.go

```diff
@@ -0,0 +1,104 @@
+package encryptionconfiguration
+
+import (
+	"crypto/rand"
+	"encoding/base64"
+	"fmt"
+	"strconv"
+	"strings"
+	"time"
+
+	apiserverconfigv1 "k8s.io/apiserver/pkg/apis/config/v1"
+)
+
+func createEncryptionKey() (apiserverconfigv1.Key, error) {
+	var key apiserverconfigv1.Key
+	keyString, err := createKeySecret()
+	if err != nil {
+		return key, err
+	}
+	if len(keyString) == 0 {
+		return key, fmt.Errorf("generated random encryption key is of unexpected length 0")
+	}
+	key.Name = createKeyName()
+	key.Secret = keyString
+	return key, nil
+}
+
+func createKeyName() string {
+	nanos := time.Now().UnixNano()
+	name := ecKeyPrefix + strconv.FormatInt(nanos, 10)
+	return name
+}
+
+func createKeySecret() (string, error) {
+	keyArray := make([]byte, ecKeySecretLen)
+	_, err := rand.Read(keyArray)
```

Comment:
Why not using a function out of https://github.com/gardener/gardener/blob/master/pkg/utils/random.go ?

Reply:
I tried to follow the recommendation in https://kubernetes.io/docs/tasks/administer-cluster/encrypt-data/ as closely as possible and first obtain full 32 Bytes of random and encode this base64 ... 

The implementation you mentioned does not match this 100% I think ...

Reply:
Hm, ok, why not? As far as I understand the secret key is just a random string, or?

Reply:
Well, the documentation states using 32 Byte keys using this approach:
head -c 32 /dev/urandom | base64

This is what I implemented. 

I could also use your recommended implementation with `GenerateRandomString` and try to figure out the `n` to use which results more or less in 32 Byte with the alphabet of your implementation (slightly reduced to the Base64 alphabet, no '+' and no `/`) or pass along the Base64 alphabet ... but I do not see that as a better option.

Reply:
OK, the documentation explicitly states 32 bytes which is nothing that the other function I linked can easily do. It would be interesting to know whether it must be exactly 32 byte, but anyway, then let's keep it as it is 👍